### PR TITLE
Update release-gh-pages.yml to use personal api token

### PR DESCRIPTION
**Pull request by worldtiki**
### :pencil: Description
- Update release-gh-pages.yml to use personal api token

### CODE REVIEW

File: .github/workflows/release-gh-pages.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ jobs:
```
Comment (ffeltrinelli) [inline .github/workflows/release-gh-pages.yml:26]: Shall we also use `persist-credentials: false` in  `actions/checkout` as suggested [here](https://github.com/cycjimmy/semantic-release-action#basic-usage)?

File: .github/workflows/release-gh-pages.yml [hunk 2]
```diff
@@ -23,7 +23,7 @@ jobs:
 
       - name: Build and push repositories json
         env:
-          GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_API_TOKEN: ${{ secrets.EG_GITHUB_TOKEN }}
```
Comment (ffeltrinelli) [inline .github/workflows/release-gh-pages.yml]: Shall we also use `EG_GITHUB_TOKEN` for `peaceiris/actions-gh-pages` below, for consistency in the entire workflow?

File: .github/workflows/release-gh-pages.yml (hunk 1 above)
Comment (worldtiki) [reply .github/workflows/release-gh-pages.yml:26]: yeap

File: .github/workflows/release-gh-pages.yml (hunk 2 above)
Comment (worldtiki) [reply .github/workflows/release-gh-pages.yml]: we can unprotect that branch which would remove the need to pass this token to that action

File: .github/workflows/release-gh-pages.yml [hunk 3]
```diff
@@ -22,8 +25,6 @@ jobs:
           npm ci
 
       - name: Build and push repositories json
-        env:
```
Comment (worldtiki) [inline .github/workflows/release-gh-pages.yml:25]: I dont think this is needed anymore

**Review by nikos912000 — APPROVED** [review]

**Pull request merged**